Tasks/Common/nuget-task-common/NuGetToolRunner2.ts

```diff
@@ -149,8 +178,25 @@ export async function getNuGetQuirksAsync(nuGetExePath: string): Promise<NuGetQu
 // identity's token.
 // Therefore, we are enabling credential provider on on-premises and disabling it on hosted (only when the version of NuGet does not support it). We allow for test
 // instances by an override variable.
+export function isCredentialProviderV1Enabled(quirks: NuGetQuirks): boolean {
+    if (quirks.hasQuirk(NuGetQuirkName.V2CredentialProvider)) {
+        tl.debug("V1 credential provider not enabled.");
+        return false;
+    }
+
+    return isCredentialProviderEnabled(quirks);
+}
+
+export function isCredentialProviderV2Enabled(quirks: NuGetQuirks): boolean {
+    if (quirks.hasQuirk(NuGetQuirkName.V2CredentialProvider) === false) {
+        tl.debug("V2 credential provider not enabled.");
+        return false;
+    }
+
+    return isCredentialProviderEnabled(quirks);
```

Comment:
No need to call this when checking V2. Let's keep them separate to avoid confusion. We should also add a build variable that lets us control if we want V2 enabled or disabled, something like `NuGet.ForceEnableCredentialProviderV2`, similar to the one we have for V1.